src/components/suggest/suggest.tsx

```diff
@@ -6,64 +6,330 @@
  */
 
-import React, { FunctionComponent } from 'react';
-import { CommonProps } from '../common';
-import { EuiSuggestItem, EuiSuggestItemProps } from './suggest_item';
-import { EuiSuggestInput, EuiSuggestInputProps } from './suggest_input';
+import React, {
+  CSSProperties,
+  FormEvent,
+  FunctionComponent,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { CommonProps, ExclusiveUnion } from '../common';
+import { useGeneratedHtmlId } from '../../services';
+
+import { EuiScreenReaderOnly } from '../accessibility';
+import { EuiFieldSearchProps } from '../form';
+import { EuiIcon } from '../icon';
+import { useEuiI18n } from '../i18n';
+import { EuiInputPopover } from '../popover';
+import { EuiSelectable, EuiSelectableListItemProps } from '../selectable';
+import { EuiToolTip } from '../tool_tip';
+
+import { EuiSuggestItem, _EuiSuggestItemPropsBase } from './suggest_item';
+import { EuiSuggestStatus, _EuiSuggestStatusMap } from './types';
+
+const statusMap: _EuiSuggestStatusMap = {
+  unsaved: {
+    icon: 'dot',
+    color: 'accent',
+  },
+  saved: {
+    icon: 'checkInCircleFilled',
+    color: 'success',
+  },
+  unchanged: {
+    icon: '',
+    color: 'success',
+  },
+  loading: {},
+};
+
+// keys of _EuiSuggestItemPropsBase
+const suggestItemPropsKeys = [
+  'label',
+  'type',
+  'description',
+  'labelDisplay',
+  'labelWidth',
+  'descriptionDisplay',
+];
+
+export interface EuiSuggestionProps
+  extends CommonProps,
+    _EuiSuggestItemPropsBase {
+  onClick?: EuiSelectableListItemProps['onClick'];
+}
+
+type _EuiSuggestProps = CommonProps &
+  EuiFieldSearchProps & {
+    /**
+     * List of suggestions to display using EuiSuggestItem.
+     * Accepts props from #EuiSuggestItemProps
+     */
+    suggestions: EuiSuggestionProps[];
 
-export type EuiSuggestProps = CommonProps &
-  EuiSuggestInputProps & {
     /**
-     * List of suggestions to display using 'suggestItem'.
+     * Changes the content of the tooltip that wraps the status icon
      */
-    suggestions: EuiSuggestItemProps[];
+    tooltipContent?: string;
 
     /**
-     * Handler for click on a suggestItem.
+     * Status of the current query 'unsaved', 'saved', 'unchanged' or 'loading'.
      */
-    onItemClick?: (item: EuiSuggestItemProps) => void;
+    status?: EuiSuggestStatus;
 
+    /**
+     * Element to be appended to the input bar.
+     */
+    append?: JSX.Element;
+
+    /**
+     * Handler for click on an EuiSuggestItem.
+     */
+    onItemClick?: (item: EuiSuggestionProps) => void;
+
+    /**
+     * Callback function called when the input changes.
+     */
     onInputChange?: (target: EventTarget) => void;
+
+    /**
+     * Callback function called when the search changes.
+     */
+    onSearchChange?: (value: string) => void;
+
+    /**
+     * Use virtualized rendering for list items with `react-window`.
+     * Best used when there are a lot of items.
+     */
+    isVirtualized?: boolean;
+    /**
```

Comment:
Same for this prop - let's add a snapshot test for it since there's some conditional logic that it affects

\+ small newline nit here
```suggestion
    isVirtualized?: boolean;

    /**
```